editor/src/components/editor/store/dispatch.tsx

```diff
@@ -742,6 +750,35 @@ function editorDispatchInner(
       }
     }
 
+    const actionNames = simpleStringifyActions(dispatchedActions)
+
+    // Check for duplicate UIDs that have originated from actions being applied.
+    const uniqueIDsResult = getAllUniqueUids(result.unpatchedEditor.projectContents)
+    if (uniqueIDsResult.duplicateIDs.length > 0) {
+      const errorMessage = `Running ${actionNames} resulted in duplicate UIDs ${JSON.stringify(
+        uniqueIDsResult.duplicateIDs,
+      )}.`
+      if (IS_TEST_ENVIRONMENT) {
+        // In tests blow out with an exception so that the error is correctly attributed.
+        throw new Error(errorMessage)
+      } else {
+        // When running in the browser log the error and tell the user to restart the editor.
+        console.error(errorMessage)
+        const errorToast = EditorActions.addToast(
+          notice(
+            `Utopia has suffered from an irrecoverable error, please reload the editor.`,
+            'ERROR',
+            true,
+            'reload-editor',
```

Comment:
can we make this error message contain the cause of the error? image trying to report or debug this based on this description

Reply:
Well we do `console.error` the detail, I'm just somewhat trying to avoid this becoming a big pile of static to a regular user.

Reply:
I don't want to block this PR since it needs to be merged, but what I'm thinking I might do after it has been merged is update this error message to say something like "Check the console for more details"